core/src/main/java/org/apache/calcite/sql2rel/RelDecorrelator.java

```diff
@@ -488,7 +488,7 @@ protected RexNode removeCorrelationExpr(
 
   public @Nullable Frame decorrelateRel(Values rel, boolean isCorVarDefined) {
     // There are no inputs, so rel does not need to be changed.
-    return null;
+    return decorrelateRel((RelNode) rel, isCorVarDefined);
```

Comment:
Perhaps we should improve the commit message and also the title of this PR because it did not clearly describe the problem